lookup_plugins/yang2spec.py

```diff
@@ -0,0 +1,1556 @@
+#
+#  Copyright 2018 Red Hat | Ansible
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+    lookup: yang2spec
+    version_added: "2.5"
+    short_description:  This plugin reads the content of given yang document and transforms it to a
+                        rules spec and configuration schema.
+    description:
+      - This plugin parses yang document and transforms it into a spec which provides a set of rules
+        This rules spec can be used to validate the input configuration to check  if it adheres
+        with respective yang model. It also outputs the  configuration schema in json format and can
+        be used as reference to build input json configuration.
+    options:
+      _terms:
+        description: The path points to the location of the top level yang module which
+        is to be transformed into to Ansible spec.
+        required: True
+      search_path:
+        description:
+          - The path is a colon (:) separated list of directories to search for imported yang modules
+            in the yang file mentioned in C(path) option. If the value is not given it will search in
+            the current directory.
+        required: false
+"""
+
+EXAMPLES = """
+- name: Get interface yang spec
+  set_fact:
+    interfaces_spec: "{{ lookup('yang2spec', 'openconfig/public/release/models/interfaces/openconfig-interfaces.yang',
+                            search_path='openconfig/public/release/models:pyang/modules/') }}"
+"""
```

Comment:
Would it be worth giving an example of how to use `interfaces_spec`?

Reply:
The example above indicates how to use yang2spec plugin. The example of how to use the output of yang2spec will be added in openconfig role